.github/workflows/ci_workflows.yml

```
@@ -108,6 +108,8 @@ jobs:
         merge-multiple: true
     - name: Upload report to Codecov
       uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
       with:
```

This is more in-sync with OpenAstronomy/github-actions-workflows#199 . Let's also skip the job if somehow we do not have any coverage file (i.e., if someday we decide to not do `-cov` jobs at all).

```suggestion
      if: ${{ hashFiles('coverage/') != '' }}
      uses: codecov/codecov-action@v4
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
```